sky/spot/recovery_strategy.py

```diff
@@ -224,15 +230,16 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
             except exceptions.ResourcesUnavailableError as e:
                 # This is raised when the launch fails after failing over
                 # through all the candidates.
-                if not any(
-                        isinstance(err, exceptions.SpotResourceUnavailableError)
-                        for err in e.failover_history):
-                    # The _launch should fail directly, if the none of the
-                    # failover candidates fails because of resource
+                if (e.failover_history and not any(
```

Comment:
Why do we need to check `e.failover_history` here? If it's empty, which iiuc means logical optimization failed to provide candidates, we will still go inside L243-245. Should document?

Reply:
Good point! That was a mistake. It is actually not needed. Removed. Thanks!